partcad/src/partcad/actions/part/convert.py

```
@@ -142,10 +152,92 @@ def update_parameters_with_defaults(part_config: dict) -> dict:
+        if "exception" in result and result["exception"] is not None:
+            pc_logging.exception(f"RenderPNG exception in render_projection: {result['exception']}")
+        pc_logging.info(f"Rendered projection (PNG) with viewport_origin={viewport_origin} to {image_path}")
+
+    except Exception as e:
+        pc_logging.error(f"Failed to render projection for {source_path.name}: {e}")
+        raise e
+
+
 def perform_conversion(project: Project, part_name: str, original_type: str,
                        part_config: dict, source_path: Path, target_format: str,
                        output_dir: Optional[str], dependencies_list: list = []) -> Path:
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)